test/conformance/enqueue/urEnqueueKernelLaunch.cpp

```diff
@@ -77,56 +77,105 @@ TEST_P(urEnqueueKernelLaunchTest, InvalidWorkDimension) {
                      UR_RESULT_ERROR_INVALID_WORK_DIMENSION);
 }
 
-struct urEnqueueKernelLaunch2DTest : uur::urKernelExecutionTest {
-    void SetUp() override {
-        program_name = "fill_2d";
-        UUR_RETURN_ON_FATAL_FAILURE(urKernelExecutionTest::SetUp());
+#define ENQUEUE_KERNEL_LAUNCH_TEST_1D_SIZES(SIZE)                              \
```

Comment:
@kbenzie let me know what you think about these added tests. Not ideal to introduce macro funcs but I think this is the most readable way to run these tests for different sizes

Reply:
GoogleTest supports parameterized tests, it turns out similarly readable and doesn't require macros like this. I'd prefer to use GoogleTest features where possible.

Reply:
Aha I will do this instead

Reply:
Here's one example of use using it for [urEnqueueUSMFill](https://github.com/oneapi-src/unified-runtime/blob/main/test/conformance/enqueue/urEnqueueUSMFill.cpp), there are others dotted around too.